dusty/compiler/spec_assembler.py

```diff
@@ -149,3 +149,16 @@ def get_all_repos(active_only=False, include_specs_repo=True):
         if spec['repo']:
             repos.add(Repo(spec['repo']))
     return repos
+
+def _get_dependent_repos_for_app_or_library(app_or_library_name):
+    specs = get_expanded_libs_specs()
+    spec = specs.get_app_or_lib(app_or_library_name)
+    repos = set()
+    for dependent_name in spec['depends']['libs']:
+        repos.add(Repo(specs.get_app_or_lib(dependent_name)['repo']))
+    return repos
+
+def get_all_repos_for_app_or_library(app_or_library_name):
```

Comment:
I feel like these functions are kind of testing-specific.  It's really getting all repos that are necessary for the test of an app or lib.  If we were getting all the repos necessary for running an app, we'd also want downstream app dependencies, for example.  If they're test specific, they should probably have different names and maybe go in a different file.  What do you think?